### PR TITLE
Correct typo and commit hash in 4.3.1 changelog

### PR DESCRIPTION
**Pull request by Juanitou**
The previous commit was leading to a 404 page in GitHub.

### CODE REVIEW

File: docs/en/04_Changelogs/4.3.1.md [hunk 1]
```diff
@@ -6,4 +6,4 @@
 
 ### Security
 
- * 2018-12-18 [c0338d191](https://github.com/silverstripe/silverstripe-framework/commit/c0338d191d8be0000ddb16b74832ed8e05ba7ff5) Fix potential SQL vulnerability in non-scalar value hyrdation (Maxime Rainville) - See [ss-2018-021](https://www.silverstripe.org/download/security-releases/ss-2018-021)
+ * 2018-12-18 [c0338d191](https://github.com/silverstripe/silverstripe-framework/commit/25bba49923a5a2cc90afcc64c58184b2fb0f2e20) Fix potential SQL vulnerability in non-scalar value hydration (Maxime Rainville) - See [ss-2018-021](https://www.silverstripe.org/download/security-releases/ss-2018-021)
```
Comment (ScopeyNZ) [inline docs/en/04_Changelogs/4.3.1.md]:
```suggestion
 * 2018-12-18 [25bba4992](https://github.com/silverstripe/silverstripe-framework/commit/25bba49923a5a2cc90afcc64c58184b2fb0f2e20) Fix potential SQL vulnerability in non-scalar value hydration (Maxime Rainville) - See [ss-2018-021](https://www.silverstripe.org/download/security-releases/ss-2018-021)
```

**Comment by ScopeyNZ** [comment]
Looks like history was rewritten at some point :suspect: 

This changelog is automatically generated so the typo exists in the commit message. I don't see the harm in updating the typo though 🙂

File: docs/en/04_Changelogs/4.3.1.md (hunk 1 above)
Comment (Juanitou) [reply docs/en/04_Changelogs/4.3.1.md]: Thanks!

**Comment by Juanitou** [comment]
> 
> 
> @Juanitou, would you like to accept @ScopeyNZ suggestion so we could merge the PR?

Sorry, I do not often use GitHub and did not see that I had to accept the commit suggestion.

**Pull request merged**